core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -268,21 +271,36 @@ object FeatureDistribution {
    */
   private def momentsValues(values: ProcessedSeq): Moments = {
     val population = values match {
-      case Left(seq) => seq.map(x => x.length.toDouble)
+      case Left(seq) => seq.map(_.length.toDouble)
       case Right(seq) => seq
     }
     MomentsGroup.sum(population.map(x => Moments(x)))
   }
 
   /**
    * Function to track frequency of the first $(MaxCardinality) unique values
-   * (number for numeric features, token for text features)
+   * (number for numeric features, token for text features). Note that shouldCleanText and tokenizeForLengths are
+   * exposed, but not relevant yet because tokenization always happens before this, during data preparation.
    *
-   * @param values          values to track distribution / frequency
-   * @return TextStats object containing a Map from a value to its frequency (histogram)
+   * @param values              values to track distribution / frequency
+   * @param shouldCleanText     whether or not to clean text for TextStats calculation
+   * @param tokenizeForLengths  whether or not to tokenize strings before computing length distribution
+   * @return TextStats object   containing a Map from a value to its frequency (histogram)
    */
-  private def cardinalityValues(values: ProcessedSeq): TextStats = {
-    TextStats(countStringValues(values.left.getOrElse(values.right.get)), Map.empty)
+  private def cardinalityValues(values: ProcessedSeq, shouldCleanText: Boolean,
+    tokenizeForLengths: Boolean): TextStats = {
+    implicit val testStatsMonoid: Monoid[TextStats] = TextStats.monoid(RawFeatureFilter.MaxCardinality)
+
+    val stringVals = values match {
+      case Left(stringSeq) => stringSeq
+      case Right(doubleSeq) => doubleSeq.map(_.toString)
+    }
+    stringVals.foldLeft(TextStats.empty)((acc, el) => acc + TextStats.textStatsFromString(
+      textString = el,
+      shouldCleanText = shouldCleanText,
+      shouldTokenize = tokenizeForLengths,
+      maxCardinality = RawFeatureFilter.MaxCardinality)
+    )
   }
 
   private def countStringValues[T](seq: Seq[T]): Map[String, Long] = {
```

Comment:
not relevant to this PR but i think `countStringValues` is no longer used.

Reply:
ok, I can remove it then